### ISSUE TITLE
 Fix process monitoring bugs in PDE

### ISSUE THREAD

**Issue by vogella** [eclipse-pde/eclipse.pde#723]
PDE contains lots of process monitoring bugs, which should be fixed.

To see them:

1.) Create .options file
2.) Add the following:

org.eclipse.equinox.common/debug=true
org.eclipse.equinox.common/progress_monitor=true

3.) Start Eclipse with the -debug flag
4.) Use PDE and check the Error log for the monitoring issue in PDE

Copied from https://bugs.eclipse.org/bugs/show_bug.cgi?id=550816

The above procedure is valid for the whole SDK.

**Comment by HannesWell** [comment]
Yes there are very likely many incorrect usages of those monitors and it will be a Sisyphean task to fix them.
Nevertheless fixes would be more than welcome.
